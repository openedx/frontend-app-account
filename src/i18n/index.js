// Placeholder be overridden by `make pull_translations`
export default {
  ar: {},
  'zh-hk': {},
  'zh-cn': {},
  uk: {},
  'tr-tr': {},
  th: {},
  te: {},
  ru: {},
  'pt-pt': {},
  'pt-br': {},
  'it-it': {},
  id: {},
  hi: {},
  he: {},
  'fr-ca': {},
  fa: {},
  'es-es': {},
  'es-419': {},
  el: {},
  'de-de': {},
  da: {},
  bo: {},
};
