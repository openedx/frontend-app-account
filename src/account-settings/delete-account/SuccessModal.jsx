import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { ModalLayer, ModalCloseButton } from '@openedx/paragon';

import messages from './messages';

export const SuccessModal = (props) => {
  const { status, intl, onClose } = props;
  return (

    <ModalLayer isOpen={status === 'deleted'} onClose={onClose}>
      <div className="mw-sm p-5 bg-white mx-auto my-3">
        <h3>
          {intl.formatMessage(messages['account.settings.delete.account.modal.after.header'])}
        </h3>
        <div className="p-3">
          <p className="h6">
            {intl.formatMessage(messages['account.settings.delete.account.modal.after.text'])}
          </p>
        </div>
        <p>
          <ModalCloseButton className="float-right" variant="link">{intl.formatMessage(messages['account.settings.delete.account.modal.after.button'])}</ModalCloseButton>
        </p>
      </div>

    </ModalLayer>

  );
};

SuccessModal.propTypes = {
  status: PropTypes.oneOf(['confirming', 'pending', 'deleted', 'failed']),
  intl: intlShape.isRequired,
  onClose: PropTypes.func.isRequired,
};

SuccessModal.defaultProps = {
  status: null,
};

export default injectIntl(SuccessModal);
