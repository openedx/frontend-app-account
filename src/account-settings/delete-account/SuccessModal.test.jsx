import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { waitFor } from '@testing-library/react';
import { SuccessModal } from './SuccessModal';

// Modal creates a portal.  Overriding createPortal allows portals to be tested in jest.
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: jest.fn(node => node), // Mock portal behavior
}));

describe('SuccessModal', () => {
  let props = {};

  beforeEach(() => {
    props = {
      onClose: jest.fn(),
      status: null,
    };
  });

  it('should match default closed success modal snapshot', async () => {
    await waitFor(() => {
      const tree = renderer.create((
        <IntlProvider locale="en"><SuccessModal {...props} /></IntlProvider>)).toJSON();
      expect(tree).toMatchSnapshot();
    });
    await waitFor(() => {
      const tree = renderer.create((
        <IntlProvider locale="en"><SuccessModal {...props} status="confirming" /></IntlProvider>)).toJSON();
      expect(tree).toMatchSnapshot();
    });
    await waitFor(() => {
      const tree = renderer.create((
        <IntlProvider locale="en"><SuccessModal {...props} status="pending" /></IntlProvider>)).toJSON();
      expect(tree).toMatchSnapshot();
    });
    await waitFor(() => {
      const tree = renderer.create((
        <IntlProvider locale="en"><SuccessModal {...props} status="failed" /></IntlProvider>)).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('should match open success modal snapshot', async () => {
    await waitFor(() => {
      const tree = renderer.create(
        <IntlProvider locale="en">
          <SuccessModal
            {...props}
            status="deleted"
          />
        </IntlProvider>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
