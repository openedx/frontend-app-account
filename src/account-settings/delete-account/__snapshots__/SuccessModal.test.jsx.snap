// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal should match default closed success modal snapshot 1`] = `null`;

exports[`SuccessModal should match default closed success modal snapshot 2`] = `null`;

exports[`SuccessModal should match default closed success modal snapshot 3`] = `null`;

exports[`SuccessModal should match default closed success modal snapshot 4`] = `null`;

exports[`SuccessModal should match open success modal snapshot 1`] = `
[
  <div
    data-focus-guard={true}
    style={
      {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={-1}
  />,
  <div
    className="pgn__modal-layer"
    data-focus-lock-disabled="disabled"
    onBlur={[Function]}
    onFocus={[Function]}
    onScrollCapture={[Function]}
    onTouchMoveCapture={[Function]}
    onWheelCapture={[Function]}
  >
    <div
      className="pgn__modal-content-container"
    >
      <div
        className="pgn__modal-backdrop"
        data-testid="modal-backdrop"
        onClick={[MockFunction]}
        onKeyDown={[MockFunction]}
      />
      <div
        className="mw-sm p-5 bg-white mx-auto my-3"
      >
        <h3>
          We're sorry to see you go!  Your account will be deleted shortly.
        </h3>
        <div
          className="p-3"
        >
          <p
            className="h6"
          >
            Account deletion, including removal from email lists, may take a few weeks to fully process through our system. If you want to opt-out of emails before then, please unsubscribe from the footer of any email.
          </p>
        </div>
        <p>
          <button
            className="pgn__modal-close-button float-right btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Close
          </button>
        </p>
      </div>
    </div>
  </div>,
  <div
    data-focus-guard={true}
    style={
      {
        "height": "0px",
        "left": "1px",
        "overflow": "hidden",
        "padding": 0,
        "position": "fixed",
        "top": "1px",
        "width": "1px",
      }
    }
    tabIndex={-1}
  />,
]
`;
